app/controllers/CategoryGuidanceController.scala

```diff
@@ -53,14 +55,22 @@ class CategoryGuidanceController @Inject() (
             categorisationInfo <- Future.fromTry(Try(CategorisationInfo.build(goodsNomenclature).get))
             updatedAnswers     <- Future.fromTry(request.userAnswers.set(CategorisationQuery, categorisationInfo))
             _                  <- sessionRepository.set(updatedAnswers)
-          } yield Ok(view())
+          } yield Ok(view(recordId))
         case None            =>
           Future.successful(Redirect(routes.JourneyRecoveryController.onPageLoad().url))
       }
   }
 
   // TODO replace index route
-  def onSubmit: Action[AnyContent] = (identify andThen getData) { implicit request =>
-    Redirect(routes.IndexController.onPageLoad.url)
+  def onSubmit(recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      auditService
```

Comment:
should be a fire and forget so don't map it just execute

Reply:
Please check now :)